service/ports/rpc/handler_blobs_get.go

```diff
@@ -1,35 +1,75 @@
 package rpc
 
 import (
+	"bytes"
 	"context"
 	"io"
 
 	"github.com/boreq/errors"
+	"github.com/planetary-social/go-ssb/service/app/queries"
 	"github.com/planetary-social/go-ssb/service/domain/messages"
 	"github.com/planetary-social/go-ssb/service/domain/transport/rpc"
 	"github.com/planetary-social/go-ssb/service/domain/transport/rpc/mux"
 )
 
-type BlobStorage interface {
-	Get() (io.ReadCloser, error)
+const (
+	// MaxBlobChunkSizeInBytes specifies the max size of one response payload
+	// when sending blobs. Larger blobs are broken into several payloads. This
+	// is done to give other pieces of code some wire time and avoid blocking
+	// the connection completely when sending the blob.
+	MaxBlobChunkSizeInBytes = 100 * 1000
+)
+
+type GetBlobQueryHandler interface {
+	Handle(query queries.GetBlob) (io.ReadCloser, error)
 }
 
 type HandlerBlobsGet struct {
+	handler GetBlobQueryHandler
 }
 
-func NewHandlerBlobsGet() *HandlerBlobsGet {
-	return &HandlerBlobsGet{}
+func NewHandlerBlobsGet(handler GetBlobQueryHandler) *HandlerBlobsGet {
+	return &HandlerBlobsGet{
+		handler: handler,
+	}
 }
 
 func (h HandlerBlobsGet) Procedure() rpc.Procedure {
 	return messages.BlobsGetProcedure
 }
 
 func (h HandlerBlobsGet) Handle(ctx context.Context, w mux.ResponseWriter, req *rpc.Request) error {
-	_, err := messages.NewBlobsGetArgumentsFromBytes(req.Arguments())
+	args, err := messages.NewBlobsGetArgumentsFromBytes(req.Arguments())
```

Comment:
As I can see we only handle the first variant of arguments that may be passed (hash only). Shouldn't we also handle the case with optional `max` and `size` args to be compliant with the spec?

Reply:
Oh yeah, that is true. I missed that.